projects/js-packages/image-guide/src/MeasurableImage.ts

```diff
@@ -83,14 +87,19 @@ export class MeasurableImage {
 	 * @param {string} url -  string The URL of the image.
 	 */
 	private async fetchFileWeight( url: string ) {
-		const response = await fetch( url, { method: 'HEAD', mode: 'no-cors' } );
+		const response = await this.fetch( url );
 		if ( ! response.url ) {
 			// eslint-disable-next-line no-console
 			console.log( `Can't get image size for ${ url } likely due to a CORS error.` );
 			return -1;
 		}
+		let size = '';
+		if ( response.headers.get( 'content-type' ).indexOf( 'text/html' ) !== -1 ) {
+			size = await response.text();
+		} else {
+			size = response.headers.get( 'content-length' );
+		}
```

Comment:
This class accepts an alternate `fetch` function, but this function treats fetch as potentially special -- not fetching the image, but returning just its size. That feels a bit off to me, like it's being treated differently by different parts of code.

When I suggested a fetch function, I had imagined that it would behave just like fetch with super-powers to bypass CORS restrictions. i.e.: fetch HEAD to get the headers, etc.

Limiting the proxy to just return file-size makes a load of sense - it's both more secure, and lighter weight than returning the full body. So I'm on the fence about what to suggest here. 

IMHO we should either:
1. Instead of accepting a fetch function, we could have the MeasurableImage class accept a getImageSize function that it calls instead of using its own logic if supplied. That way, we could put the special-case logic in the getImageSize call, and *expect* just a size in the body, or
2. Make the fetch function behave like fetch, and just fetch the HEAD.

What do you think?

Reply:
I realised after working on this for a while that I was returning the content length in the ajax request body, and so it wouldn't be a drop-in replacement for fetch. It's impossible to set the content-length header of the ajax request to be the length of the proxied image. You have to pass the content length in the body.

While I modified the function so it returns the entire header array, I think I should just go with a drop-in getImageSize function instead.